lnd_test.go

```diff
@@ -3801,69 +3778,110 @@ func testSpiderShortestPath(net *lntest.NetworkHarness, t *harnessTest) {
 				nodeNames[connections[i][1]], nodeNames[connections[i][0]], err)
 		}
 	}
-	
 
-	// Open up a payment streams to each node, that we'll use to
-	// send payment between nodes.
 
-	payStreams := make([]lnrpc.Lightning_SendPaymentClient, numNodes)
+	timeLockDelta := uint32(144)
+
+	expectedPolicy := &lnrpc.RoutingPolicy{
+		FeeBaseMsat:      baseFee,
+		FeeRateMilliMsat: feeRate * testFeeBase,
+		TimeLockDelta:    timeLockDelta,
+		MinHtlc:          1000, // default value
+	}
 
+	graphSubs := make([]graphSubscription, numNodes)
 	for i := 0; i < numNodes; i++ {
-		ps, err := nodes[i].SendPayment(ctxb)
-		payStreams[i] = ps
-		if err != nil {
-			t.Fatalf("unable to create payment stream for %v: %v", nodeNames[i], err)
-		}
+		ctxt, _ := context.WithTimeout(ctxb, timeout)
+		graphSubs[i] = subscribeGraphNotifications(t, ctxt, nodes[i])
+		defer close(graphSubs[i].quit)
 	}
 
-	// send payments
-	for i := 0; i < numInvoices; i++ {
-		for j := 0; j < numPayIntents; j++ {
-			sendReq := &lnrpc.SendRequest{
-				PaymentRequest: invoices[j][i],
-				SpiderAlgo: routing.ShortestPath,
-			}
-			if err := payStreams[payIntents[j][0]].Send(sendReq); err != nil {
-				t.Fatalf("unable to send payment: "+
-					"%v", err)
-			}
+	expectedMessages := make([]expectedChanUpdate, numChannels)
+	for i := 0; i < numChannels; i++ {
+		expectedMessages[i] = expectedChanUpdate{
+			nodes[connections[i][0]].PubKeyStr, expectedPolicy, chanPoints[i]}
+	} 
+
```

Comment:
Can you add a brief comment on what exactly is happening here? In particular, why do we need to "update" the policy?